crates/astria-sequencer/src/checked_actions/ibc_relay.rs

```diff
@@ -80,6 +95,51 @@ impl CheckedIbcRelay {
                 .wrap_err("failed to check if address is IBC relayer")?,
             "transaction signer not authorized to execute IBC actions"
         );
+
+        // Only allow ics20 transfers of fee assets post-Blackburn upgrade
+        if !use_pre_blackburn_ics20_transfer(&state)
+            .await
+            .wrap_err("failed to get blackburn upgrade status")?
+        {
+            if let IbcRelay::RecvPacket(msg_recv_packet) = &self.action {
```

Comment:
This block looks correct to me, seems like a lot of code reuse from the actual receive on the action? Maybe the upgrade logic should be in the time of execution flow as well?

https://github.com/astriaorg/astria/blob/63e0c3c42f383f5f3513166cd3cabf909f7e96d4/crates/astria-sequencer/src/ibc/ics20_transfer.rs#L445

Reply:
We should be safe on that front, in that the code here is called not only when we try to construct a `CheckedIbcRelay`, but also right at the start of execution, so essentially it is in the execution flow.